spec/features/users_spec.rb

```diff
@@ -0,0 +1,148 @@
+require "rails_helper"
+
+describe UsersController do
+  feature "user can visit the new user page" do
+    scenario "when arriving on the welcome page" do
+      visit root_path
+      click_button "New Account"
+
+      expect(page).to have_text("Please fill in the form to sign up")
+    end
+  end
+
+  feature "can sign up for an account" do
+    scenario "with valid attributes" do
+      visit new_user_path
+
+      within "form" do
+        fill_in "Name", with: "Mary"
+        fill_in "Email", with: "Mary@example.com"
+        fill_in "Password", with: "secret"
+        click_button "Sign up"
+      end
+
+      expect(page).to have_text("Welcome Mary")
+    end
+  end
+
+  feature "can't sign up for an account" do
+    scenario "without a name" do
+      visit new_user_path
+
+      within "form" do
+        fill_in "Email", with: "Mary@example.com"
+        fill_in "Password", with: "secret"
+        click_button "Sign up"
+      end
+
+      expect(page).to have_text("User could not be created")
+      expect(page).to have_text("Please fill in the form to sign up")
+    end
+
+    scenario "without an email" do
+      visit new_user_path
+
+      within "form" do
+        fill_in "Name", with: "Mary"
+        fill_in "Password", with: "secret"
+        click_button "Sign up"
+      end
+
+      expect(page).to have_text("User could not be created")
+      expect(page).to have_text("Please fill in the form to sign up")
+    end
+
+    scenario "without a password" do
+      visit new_user_path
+
+      within "form" do
+        fill_in "Name", with: "Mary"
+        fill_in "Email", with: "Mary@example.com"
+        click_button "Sign up"
+      end
+
+      expect(page).to have_text("User could not be created")
+      expect(page).to have_text("Please fill in the form to sign up")
+    end
+  end
+
+  feature "user can see their show page" do
+    scenario "signing up with valid attributes" do
+      user = create_user("John", "John@example.com", "password")
+      visit user_path(user)
+
+      expect(page).to have_text("Welcome John")
+    end
+  end
+
+  feature "user can visit the user edit page" do
+    scenario "logging in with valid attributes" do
+      user = create_user("John", "John@example.com", "password")
+      visit edit_user_path(user)
+
+      expect(page).to have_text("Please fill in the form to edit your profile")
+    end
+  end
+
+  feature "user can update their profile" do
+    scenario "logging in with valid attributes" do
+      user = create_user("John", "John@example.com", "password")
+      visit edit_user_path(user)
+
+      within "form" do
+        fill_in "Email", with: "JT@example.com"
+        click_button "Edit profile"
+      end
+
+      expect(page).to have_text("Your email address is: JT@example.com")
+    end
+  end
+
+  feature "user can't update their profile" do
```

Comment:
A suggestion to move these specs to a controller spec

Reply:
Ok, will implement a controller spec and move these with the others, thanks Elle!